### PR TITLE
Fix issues OP-70 and OP-73

### PR DESCRIPTION
**Pull request by rak16**
Bug Number: [OP-70](https://project-sunbird.atlassian.net/browse/OP-70), [OP-73](https://project-sunbird.atlassian.net/browse/OP-73)

Issues: 
- Cloud search doesn't work when the device is booted up with no internet connectivity
- Cloud search fails to work if any error occurs at the time of fetching search keys

RCA: The fetching of keys was happening only **once** when `devmgmt` service got started and there was no mechanism to re-fetch the keys. Due to this
- if there was no internet connectivity the request would time-out and there will no attempt to request the keys again
- if any error occured at the time of fetching search keys, again, there will be no attempt to re-fetch the keys

Fix: Check for Internet connectivity before fetching the keys and add mechanism to re-fetch keys in case of an error while fetching keys/performing search.

### CODE REVIEW

File: devmgmtV2/controllers/cloud.controller.js [hunk 1]
```diff
@@ -172,22 +193,39 @@ let searchContent = (req, res) => {
...
+				} else {
+					console.log("Generating cloud auth token...");
+					return generateOriginalJWTs();
+				}
+			} else {
+				throw new Error("No Internet connectivity.");
 			}
+		})
+		.then(({ token }) => {
+			state.authToken(token);
```
Comment (chetansk) [inline devmgmtV2/controllers/cloud.controller.js]: You do not need to check the validity of the token here ?

File: devmgmtV2/controllers/cloud.controller.js [hunk 2]
```diff
@@ -16,7 +18,7 @@ let {
 	generateOriginalJWTs
 } = require('../helpers/cloud.helper.js');
 
-let getState = () => {
+let state = (() => {
```
[the remark this answers is not in the document]
Comment (rak16) [reply devmgmtV2/controllers/cloud.controller.js]: Yes. Earlier `getState` was getting invoked manually and returned value (ie. the state) was being stored in a variable. Now, the function is immediately invoked and the returned value is assigned to `state`. So `getState` doesn't make sense.

File: devmgmtV2/controllers/cloud.controller.js [hunk 3]
```diff
@@ -48,10 +57,26 @@ let getState = () => {
...
+let getInternetStatus = () => {
+	let defer = q.defer();
+
+    const cmd = path.join(__dirname, '../../CDN/netconnect_status.sh');
+
+    exec(cmd, { shell: '/bin/bash' }, (err, stdout, stderr) => {
+    	if(err) {
+            defer.resolve(false);
+    	}  else {
+            defer.resolve(!!stdout.trim());
```
Comment (chetansk) [inline devmgmtV2/controllers/cloud.controller.js]: Please use the return value, rather than stdout

**Comment by chetansk** [comment]
Please fix one comment and commit the code

**Pull request merged**